pkg/lib/flavor/util/platform_flavor_util.go

```diff
@@ -210,33 +211,38 @@ func (pfutil PlatformFlavorUtil) GetHardwareSectionDetails(hostInfo *taModel.Hos
 		// Set TPM Feature presence
 		tpm := cm.TPM{}
 		tpm.Enabled = hostInfo.HardwareFeatures.TPM.Enabled
-		tpm.Enabled = hostInfo.HardwareFeatures.TPM.Enabled
-		tpm.Version = hostInfo.HardwareFeatures.TPM.Meta.TPMVersion
+		tpm.Meta.TPMVersion = hostInfo.HardwareFeatures.TPM.Meta.TPMVersion
 		// split into list
-		tpm.PcrBanks = strings.Split(hostInfo.HardwareFeatures.TPM.Meta.PCRBanks, constants.PCRBankSeparator)
-		feature.TPM = &tpm
-
-		txt := cm.TXT{}
-		if hostInfo.HardwareFeatures.TXT != nil {
-			// Set TXT Feature presence
-			txt.Enabled = hostInfo.HardwareFeatures.TXT.Enabled
-			feature.TXT = &txt
-		}
+		tpm.Meta.PCRBanks = strings.Split(hostInfo.HardwareFeatures.TPM.Meta.PCRBanks, constants.PCRBankSeparator)
+		feature.TPM = tpm
+
+		txt := cm.HardwareFeature{}
+		// Set TXT Feature presence
+		txt.Enabled = hostInfo.HardwareFeatures.TXT.Enabled
```

Comment:
what if TXT details are missing?

Reply:
As per new design, all the details will be provided in platform info. Followed the same for host manifest and flavor. Will that be okay ?

Reply:
I don't think that should be ok. As there is a difference between the following steps,

1. Feature Support Missing - if details are missing
2. Feature is enabled
3. Feature is disabled